inbm/dispatcher-agent/dispatcher/schedule/manifest_parser.py

```diff
@@ -76,9 +76,12 @@ def _get_schedules(self) -> None:
             if hasattr(update_schedule, 'schedule'):
                 schedule = update_schedule.schedule
                 if 'single_schedule' in schedule:
-                    self._parse_single_schedule(schedule, manifests, request_id)
+                    schedule_details = Schedule(request_id=request_id, manifests=manifests)
+                    self._parse_single_schedule(schedule, schedule_details)
                 if 'repeated_schedule' in schedule:
-                    self._parse_repeated_schedule(schedule, manifests, request_id)
+                    schedule_details = Schedule(request_id=request_id, manifests=manifests)
+                    self._parse_repeated_schedule(schedule, schedule_details)
```

Comment:
**<h2>$\color{#FFA500}Improvements: Code Refactoring$</h2>**
**Suggestion:** To enhance code readability and maintainability, consider extracting the logic for parsing single and repeated schedules into separate methods.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
                self._process_schedule_types(schedule, request_id, manifests)
```